File: src/main/java/iot/tetracubered/iotMessaging/deviceProvisioning/payloads/DeviceProvisioningPayload.java
package iot.tetracubered.iotMessaging.deviceProvisioning.payloads;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.UUID;

public class DeviceProvisioningPayload {

    private final UUID id;
    private final String name;
    private final String feedbackTopic;
    private final List<DeviceFeatureProvisioningPayload> features;

    @JsonCreator
    public DeviceProvisioningPayload(@JsonProperty("id") UUID id,
                                     @JsonProperty("name") String name,
                                     @JsonProperty("feedback-topic") String feedbackTopic,
                                     @JsonProperty("features") List<DeviceFeatureProvisioningPayload> features) {
        this.id = id;
        this.name = name;
        this.feedbackTopic = feedbackTopic;
        this.features = features;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getFeedbackTopic() {
        return feedbackTopic;
    }

    public List<DeviceFeatureProvisioningPayload> getFeatures() {
        return features;
    }
}

File: src/main/kotlin/iot/tetracubered/iot/payloads/DeviceProvisioningPayload.kt
package iot.tetracubered.iot.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

data class DeviceProvisioningPayload @JsonCreator constructor(
    @field:JsonProperty("id") val id: UUID,
    @field:JsonProperty("name") val name: String,
    @field:JsonProperty("feedback_topic") val feedbackTopic: String,
    @field:JsonProperty("features") val features: List<DeviceFeatureProvisioningPayload>
)